File: scrape.py
import urllib.request
import json
from bs4 import BeautifulSoup

try:
    req = urllib.request.Request('https://dashebeauty.com/shop/', headers={'User-Agent': 'Mozilla/5.0'})
    html = urllib.request.urlopen(req).read().decode('utf-8')
    soup = BeautifulSoup(html, 'html.parser')
    products = soup.select('li.product')
    
    scraped = []
    for p in products[:8]:
        title_el = p.select_one('.woocommerce-loop-product__title')
        price_el = p.select_one('.woocommerce-Price-amount bdi')
        img_el = p.select_one('img')
        link_el = p.select_one('a.woocommerce-LoopProduct-link')
        
        title = title_el.text.strip() if title_el else 'N/A'
        price = price_el.text.strip() if price_el else 'N/A'
        img = img_el.get('src') if img_el else 'N/A'
        link = link_el.get('href') if link_el else 'N/A'
        
        scraped.append({
            'title': title,
            'price': price,
            'image': img,
            'link': link
        })
        
    print(json.dumps(scraped, indent=2))
except Exception as e:
    print('Error:', e)

File: scrape.js
const https = require('https');

https.get('https://dashebeauty.com/shop/', (resp) => {
  let data = '';
  resp.on('data', (chunk) => { data += chunk; });
  resp.on('end', () => {
    // Regex to match li.product
    const products = [];
    const productRegex = /<li[^>]*class="[^"]*product[^"]*"[^>]*>([\s\S]*?)<\/li>/gi;
    let match;
    while ((match = productRegex.exec(data)) !== null && products.length < 8) {
      const pData = match[1];
      
      const titleMatch = pData.match(/<h2[^>]*woocommerce-loop-product__title[^>]*>([^<]+)<\/h2>/i);
      const title = titleMatch ? titleMatch[1].trim() : 'Unknown';
      
      const priceMatch = pData.match(/<span class="woocommerce-Price-amount amount"><bdi>(?:<span[^>]*>[^<]+<\/span>)?([\d.,]+)/i);
      const price = priceMatch ? priceMatch[1].trim() : '0.000';
      
      const imgMatch = pData.match(/<img[^>]*src="([^"]+)"/i);
      const img = imgMatch ? imgMatch[1] : '';
      
      products.push({ title, price, image: img });
    }
    console.log(JSON.stringify(products, null, 2));
  });
}).on("error", (err) => {
  console.log("Error: " + err.message);
});
